keyring/cli.py

```diff
@@ -41,7 +42,7 @@ def __init__(self):
         self.parser.add_argument(
             "--disable", action="store_true", help="Disable keyring and exit"
         )
-        self.parser._operations = ["get", "set", "del", "diagnose"]
+        self.parser._operations = ["get", "set", "del", "getcreds", "diagnose"]
```

Comment:
When naming things, I tend to separate words with something (space, dash, underscore). Here, I think dash may be appropriate.
```suggestion
        self.parser._operations = ["get", "set", "del", "get-creds", "diagnose"]
```

But then I notice we have `get` and `get-creds` and the asymetry is bothering me, and it makes me wonder if it should be `keyring get password` + `keyring get credential` (or `get pass` and `get creds`) or maybe `get` with `--creds` or `--password` (default). All of those options have more onerous transitional concerns, so I'm not loving any option.

I don't feel strongly about it, so happy to move forward with something pragmatic.

Reply:
`get-creds` seems like simplest transition for the API, I feel your pain though. Perhaps it'd be worth opening a tracking issue to consider a different API in some future major version. I think `get` with some sort of `--mode` might make the most sense. You could also just assume a single value is the service name and that the username is null and return the full credentials in that case? eek.

Reply:
> I think get with some sort of --mode might make the most sense.

This would be my personal vote over separate command.  Updated PR to add `--get-mode` with default `password`

Help text for reference:
```
  --get-mode {password,creds}
                        Mode for 'get' operation. 'password' requires a username and will return only the password.
                        'creds' does not require a username and will return both the username and password separated by
                        a newline. Default is 'password'
```

Reply:
I don't really understand how we're going to use this downstream though. We'll need to try try `--get-mode creds` first and if it fails fall back to excluding the flag? Presumably there will be a different exit code? I guess we'll cache this so we don't need to do it on every request? but we have a lot of concurrent requests that make this difficult to do efficiently.

Reply:
If no username on the index url, use `--get-mode creds` else use default.  After that, the cacheing by netloc should just work

Reply:
Yeah but we won't know if the user's `keyring` version supports this without trial and error.

Reply:
> Yeah but we won't know if the user's keyring version supports this without trial and error.

You can tell this is the problem on the first call if the exit code is `2` instead of `1` and don't try again after first `2` status.

```
➜ keyring get --get-mode creds https://example.com/ oauth2accesstoken
usage: keyring [-h] [-p KEYRING_PATH] [-b KEYRING_BACKEND] [--list-backends]
               [--disable] [--print-completion {bash,zsh,tcsh}]
               [{get,set,del,diagnose}] [service] [username]
keyring: error: unrecognized arguments: --get-mode creds https://example.com/
X➜   echo $?                       +
2
➜  keyring get https://example.com/ oauth2accesstoken
X➜  echo $?                       +
1
```

The user manually requests `keyring` use, so `uv` would already be failing for them if no username provided.  This is just a different failure mode.

Reply:
Err... I think I see an issue - for index-urls which don't need keyring (PyPi) - you'd hit the status code `2` when keyring requested but don't know whether you need to warn the user about it.

Reply:
> I guess we'll cache this so we don't need to do it on every request? but we have a lot of concurrent requests that make this difficult to do efficiently.

On this point, you're using a mutex on the keyring cache, so the first time a host fails the check, you're returning None for all subsequent checks on that host.  The only way you'll run into new efficiency issues is if there are a large number of unique index-url hosts involved.  This seems unlikely.

Reply:
👍  Sorry I didn't mean to distract, unless usage in `uv` helps inform the design here I don't think we need to discuss it too much.

Reply:
@zanieb We're kind of off-topic to discussion of this PR - can move to draft example PR on `uv` here:
https://github.com/astral-sh/uv/pull/3081